File: Dummy/hawkshaw/library/datalayer/network/grpc/interceptors/AuthInterceptor.java
package com.hawkshaw.library.datalayer.network.grpc.interceptors;

import Y1.C0106f;
import Y1.C0108g;
import Y1.C0111h0;
import Y1.C0114j;
import Y1.C0115j0;
import Y1.C0116k;
import Y1.m0;
import Y1.p0;
import com.hawkshaw.library.BuildConfig;
import d1.e;
import java.util.BitSet;

public final class AuthInterceptor implements C0116k {
    /* access modifiers changed from: private */
    public final m0 baseHeaders;
    /* access modifiers changed from: private */
    public final String tokenScheme = "Bearer";

    /* JADX WARNING: type inference failed for: r0v1, types: [Y1.m0, java.lang.Object] */
    public AuthInterceptor() {
        ? obj = new Object();
        e eVar = m0.f2789d;
        BitSet bitSet = C0115j0.f2781d;
        obj.e(new C0111h0("x-version-code", eVar), "109");
        obj.e(new C0111h0("x-version-name", eVar), BuildConfig.VERSION_NAME);
        this.baseHeaders = obj;
    }

    public <ReqT, RespT> C0114j interceptCall(p0 p0Var, C0106f fVar, C0108g gVar) {
        return new AuthInterceptor$interceptCall$1(this, gVar != null ? gVar.h(p0Var, fVar) : null);
    }
}

File: app/src/main/java/com/hawkshaw/library/datalayer/network/grpc/interceptors/AuthInterceptor.kt
package com.hawkshaw.library.datalayer.network.grpc.interceptors

import io.grpc.CallOptions
import io.grpc.Channel
import io.grpc.ClientCall
import io.grpc.ClientInterceptor
import io.grpc.ForwardingClientCall
import io.grpc.Metadata
import io.grpc.MethodDescriptor
import com.hawkshaw.app.BuildConfig

/**
 * A gRPC interceptor that adds authentication and application version headers to outgoing requests
 */
class AuthInterceptor : ClientInterceptor {
    private val tokenScheme = "Bearer"
    private val baseHeaders = Metadata().apply {
        put(
            Metadata.Key.of("x-version-code", Metadata.ASCII_STRING_MARSHALLER),
            "109"
        )
        put(
            Metadata.Key.of("x-version-name", Metadata.ASCII_STRING_MARSHALLER),
            try {
                BuildConfig.VERSION_NAME
            } catch (e: Exception) {
                "1.0.0" // Fallback version
            }
        )
    }

    override fun <ReqT, RespT> interceptCall(
        method: MethodDescriptor<ReqT, RespT>,
        callOptions: CallOptions,
        next: Channel
    ): ClientCall<ReqT, RespT> {
        // Forward the call with added authentication headers
        return object : ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(
            next.newCall(method, callOptions)
        ) {
            override fun start(responseListener: Listener<RespT>, headers: Metadata) {
                // Add base headers that include version information
                headers.merge(baseHeaders)
                
                // Add authentication token if available
                // Note: In real implementation, this would get the token from some secure storage
                // val token = getTokenFromStorage()
                // if (token != null) {
                //     headers.put(
                //         Metadata.Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER),
                //         "$tokenScheme $token"
                //     )
                // }
                
                super.start(responseListener, headers)
            }
        }
    }
} 